lib/shared/inference/function_list.rb

```diff
@@ -50,7 +50,12 @@ def coerce_arguments!(parsed)
 
             type = parameter[:type]
             if type == "array"
-              arguments[name] = JSON.parse(value)
+              arguments[name] = begin
+                JSON.parse(value)
+              rescue StandardError
+              end
+              # TODO consider other heuristics as well
```

Comment:
IME a `# TODO` with no name never gets done :) Can you please change to `TODO (sam)` so it has an owner?

Reply:
Also what other heuristics are we talking about?

Reply:
thanks, yeah will improve, problem is LLM returns an "array" but you have no idea how it made the array. 

trying json first is good, but then maybe splitting on new line , splitting on comma , deciding what to do based on length / regex... its a tough one.

Reply:
Ugh yeah that's annoying that it doesn't have a consistent response...